torchgeo/datasets/splits.py

```diff
@@ -281,7 +281,8 @@ def roi_split(dataset: GeoDataset, rois: Sequence[Polygon]) -> list[GeoDataset]:
 
 
 def time_series_split(
-    dataset: GeoDataset, lengths: Sequence[float | pd.Timedelta | pd.Interval]
+    dataset: GeoDataset,
+    lengths: Sequence[float] | Sequence[Timedelta] | Sequence[Interval[Timestamp]],
```

Comment:
I think this is actually a bug. Previously, we allowed any `Sequence[a, b, c]`, which I think allows mixing a, b, and c in a single list. In fact, we only support pure lists of a, b, or c.